distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/UrlResourceReaderConfigurator.java

```diff
@@ -30,38 +34,94 @@
 
   private static final String PID = "ddf.catalog.resource.impl.URLResourceReader";
 
+  private static final String ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY = "rootResourceDirectories";
+
   private ConfigurationAdmin configAdmin;
 
   public UrlResourceReaderConfigurator(ConfigurationAdmin configAdmin) {
     this.configAdmin = configAdmin;
   }
 
   public void setUrlResourceReaderRootDirs(String... rootResourceDirs) throws IOException {
-    Configuration configuration = configAdmin.getConfiguration(PID, null);
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    Set<String> rootResourceDirectories =
+    Collection<String> newRootResourceDirs =
         ImmutableSet.<String>builder().add(rootResourceDirs).build();
-    properties.put("rootResourceDirectories", rootResourceDirectories);
-    configuration.update(properties);
-    for (int i = 0; i < 5; i++) {
-      Configuration updatedConfig = configAdmin.getConfiguration(PID, null);
-      if (updatedConfig
-          .getProperties()
-          .get("rootResourceDirectories")
-          .equals(rootResourceDirectories)) {
-        break;
-      }
-      boolean interrupted = false;
-      try {
-        TimeUnit.SECONDS.sleep(5);
-      } catch (InterruptedException e) {
-        interrupted = true;
-      } finally {
-        if (interrupted) {
-          Thread.currentThread().interrupt();
-        }
+    Configuration configuration = configAdmin.getConfiguration(PID, null);
+    Optional<Collection<String>> currentRootResourceDirs =
+        getCurrentRootResourceDirs(configuration);
+
+    if (currentRootResourceDirs.isPresent()) {
```

Comment:
✏️ I am not sure there was any value in having `getCurrentRootResourceDirs()` return an `Optional` since you are checking its presence and retrieving its value twice.